src/realm/query_expression.cpp

```diff
@@ -519,11 +519,16 @@ void LinkCount::evaluate(size_t index, ValueBase& destination)
                 const Obj obj = m_link_map.get_target_table()->get_object(links[i]);
                 auto val = obj._get<int64_t>(m_column_key.get_index());
                 size_t s;
-                if (val & 1) {
+                if (m_column_key.get_type() == col_type_Link && !m_column_key.is_collection()) {
+                    // It is a single link column
+                    s = (val == 0) ? 0 : 1;
+                }
+                else if (val & 1) {
```

Comment:
How can you be sure that, at this point, we are dealing with a simple backlink column that points to one value? I guess what I am trying to say, is that... the logic behind `val == 0` vs `val == 1` is a little bit obscure, does `val` represents a simple boolean flag?....

Reply:
Maybe a bit obscure. The only case where we can have a tagged value here is when we are dealing with a backlink column with a single link. All other cases we have a ref to bplustree of links.

Reply:
OK maybe we can use a comment here. Describing this.

Reply:
There is actually a comment